### DISCUSSION TITLE
Fetch multiple files content in one api

### DISCUSSION THREAD

**Discussion by emadimehrdad** [python-gitlab/python-gitlab#2434] [Q&A]
Hello
I want to get the contents of all files in a folder or in repo but I didn't find anything. The one way I find is to get all files' names from repository_tree and get every single file content from "project files" API. 
I want to get content from multiple files at the same time with one API.
Is there any way?
I've searched a lot in documents but didn't find the answer to my question.

**Comment by nejch** [comment]
You can use the file archive API (`project.repository_archive()`), which supports a path argument. Upstream docs are in https://docs.gitlab.com/ee/api/repositories.html#get-file-archive

From our docs (https://python-gitlab.readthedocs.io/en/stable/gl_objects/projects.html):
```python
tgz = project.repository_archive()

# get the archive for a branch/tag/commit
tgz = project.repository_archive(sha='4567abc')

# get the archive in a different format
zip = project.repository_archive(format='zip')
```

**Reply by emadimehrdad** [reply]
Many thanks.
Can not we use the path argument in the python-gitLab repository?
I guess it is not implemented.
